vortex-array/src/array/sparse/compute.rs

```diff
@@ -116,3 +124,142 @@ impl ScalarAtFn for SparseArray {
         }
     }
 }
+
+impl TakeFn for SparseArray {
+    fn take(&self, indices: &dyn Array) -> VortexResult<ArrayRef> {
+        let flat_indices = flatten_primitive(indices)?;
+        // if we are taking a lot of values we should build a hashmap
+        // TODO(robert): Figure out at which point hashmap is slower than searching
+        let exact_taken_indices = if indices.len() > 1_000 && indices.len() * 100 > self.len() {
```

Comment:
I think this biases too strongly toward binary search... my `take_10K` benchmark (out of 1M) would just barely fail to satisfy this condition, despite seeing an -87% in runtime with hash map

Reply:
Do you want to give me better numbers?

Reply:
I can also write a benchmark and find it

Reply:
Anyway though it's easy to tweak these numbers once you have both variants

Reply:
yeah, we can merge this, and then I can use my benchmark to tweak/tune. fwiw, my benchmark only had 100 patches out of ~1M records (taking 10K), but those 10K binary searches were basically 1.2ms... so 120ns per binary search of 100 indices.

Meanwhile, nearly same benchmark but with 0 patches took 120 micros less... so that's roughly the cost of hashmap building + querying + applying patches.

So, it's just one run of the benchmark, caveats apply, etc. But if we assume that 120 micros is *entirely* spent on the fixed overhead of building the map (i.e., the allocation), then it's a breakeven at ~1000. 

That's pretty conservative, so I'm betting the right condition is literally going to be in the ballpark of `if indices.len() > 512`

Reply:
cool, I updated the threshold